packages/visualizations/src/components/utils/ColorsLegend.svelte

```diff
@@ -1,13 +1,47 @@
 <svelte:options immutable={true} />
 
 <script lang="ts">
+    import { onDestroy, onMount } from 'svelte';
+    import { debounce } from 'lodash';
     import type { DataBounds, ColorsScale, LegendVariant } from '../types';
     import { defaultCompactLegendNumberFormat } from './formatter';
+
     // options to customize the component
     export let dataBounds: DataBounds;
     export let colorsScale: ColorsScale;
     export let variant: LegendVariant;
     export let title: string;
+
+    // the part below is related to labels rotation
+    let legendWidth: number;
+    let colorBoxWidth: number;
+    const labelsWidth: number[] = [];
+    let maxLabelsSize: number;
+    let numberOfLabels: number;
+    let displayVertical: boolean;
+    const handleLabelRotation = (): void => {
```

Comment:
Here is a working exemple: 
```suggestion
        const handleLabelRotation = (legendW: number, labelW: number[]): void => {
        if (colorsScale.type === 'palette' && labelsWidth.length !== 0 && legendWidth) {
            console.log();
            const availableWidthPerLabel: number = legendWidth / numberOfLabels - 3;
            colorBoxWidth = legendWidth / numberOfLabels;
            numberOfLabels = colorsScale.colors.length + 1;
            maxLabelsSize = Math.max(...labelsWidth);
            return availableWidthPerLabel < maxLabelsSize;
        }
    };

    $: displayVertical = handleLabelRotation(legendWidth, labelsWidth);
```

What I like better in this pattern is that you have what triggers reactivity as argument and what is updated during the cycle on the left-handside.

I don't think debouncing is problem in itself as long as it has `legendWidth` as parameter and truly debounces the resize and is not used defensively against Svelte's cycles because we are triggering reactivity on the wrong variables.